src/ui/components/filters/sortoptionscomponent.js

```diff
@@ -43,6 +43,33 @@ export default class SortOptionsComponent extends Component {
         }
       }
     });
+
+    this.core.storage.registerListener({
+      eventType: 'update',
+      storageKey: StorageKeys.HISTORY_POP_STATE,
+      callback: () => {
+        const persistedOptionIndex = this.getPersistedOptionIndex();
+        this._updateSelectedOption(persistedOptionIndex);
+        this.setState();
+      }
+    });
+  }
+
+  /**
+   * Returns the option index matching the persisted sortBys, if one exists.
+   *
+   * @returns {number|undefined}
+   */
+  getPersistedOptionIndex () {
```

Comment:
To clarify, this function is returning the selected persisted option, correct? If so, what do you think of renaming this to `getSelectedPersistedOptionIndex`?

Reply:
Technically the option isn't selected yet though. It seems a little weird to call it that when the method has nothing to do with whether the option is selected or not

Reply:
ah, I see. But we do only persist the selected options, right?

Reply:
Ah yeah that's true. What do you think about `getPersistedSelectedOptionIndex` then?

Reply:
that works!